eis_toolkit/exploratory_analyses/statistical_tests.py

```diff
@@ -52,27 +53,64 @@ def chi_square_test(data: pd.DataFrame, target_column: str, columns: Optional[Se
 
 
 @beartype
-def normality_test(data: pd.DataFrame) -> dict:
+def normality_test(
+    data: Union[pd.DataFrame, np.ndarray], columns: Optional[Sequence[str]] = None
+) -> Union[dict, Tuple]:
     """Compute Shapiro-Wilk test for normality on the input data.
 
-    It is assumed that the input data is normally distributed and numeric, i.e. integers or floats.
-
     Args:
-        data: Dataframe containing the input data.
+        data: Dataframe or numpy array containing the input data.
+        columns: Optional Columns to be used for testing.
 
     Returns:
         Test statistics for each variable.
 
     Raises:
-        EmptyDataFrameException: The input Dataframe is empty.
+        EmptyDataException: The input data is empty.
+        InvalidColumnException: All selected columns were not found in the input data.
+        NonNumericDataException: Selected data or columns contains non-numeric data.
+        SampleSizeExceededException: Input data exceeds the maximum of 5000 samples.
     """
-    if check_empty_dataframe(data):
-        raise exceptions.EmptyDataFrameException("The input Dataframe is empty.")
-
     statistics = {}
-    for column in data.columns:
-        statistic, p_value = shapiro(data[column])
-        statistics[column] = (statistic, p_value)
+    if isinstance(data, pd.DataFrame):
+        if check_empty_dataframe(data):
+            raise exceptions.EmptyDataException("The input Dataframe is empty.")
+
+        if columns is not None:
+            if not check_columns_valid(data, columns):
+                raise exceptions.InvalidColumnException("All selected columns were not found in the input DataFrame.")
+            if not check_columns_numeric(data, columns):
+                raise exceptions.NonNumericDataException("The selected columns contain non-numeric data.")
+
+            data = data[columns].dropna()
+
+        else:
+            if not check_columns_numeric(data, data.columns):
+                raise exceptions.NonNumericDataException("The input data contain non-numeric data.")
+            columns = data.columns
+
+        for column in columns:
+            if len(data[column]) > 5000:
+                raise exceptions.SampleSizeExceededException(
+                    f"Sample size for '{column}' exceeds the limit of 5000 samples."
+                )
+            statistic, p_value = shapiro(data[column])
+            statistics[column] = (statistic, p_value)
+
+    else:
+        if data.size == 0:
+            raise exceptions.EmptyDataException("The input numpy array is empty.")
+        if not np.issubdtype(data.dtype, np.number):
+            raise exceptions.NonNumericDataException("The input data contain non-numeric data.")
```

Comment:
I would remove this since it should not be possible for Numpy data to have non-numeric data. We also don't test this in other functions using Numpy data so it is safe to assume we are dealing with numeric data.